dask/imperative.py

```diff
@@ -114,11 +114,18 @@ def applyfunc(func, args, kwargs, pure=False):
     of that computation."""
 
     args, dasks = unzip(map(to_task_dasks, args), 2)
+    g = groupby(lambda v: hasattr(v[1], 'dask'), kwargs.items())
```

Comment:
Shouldn't we recurse down kwargs.values?

E.g. can we add a test for `f(key=[mydaskobj])`